File: LABORATORY/laboratory1.py
import os
import subprocess
import sys

pid = 1

try:
    p = subprocess.Popen(["whoami"])
except:
    pid = -1

if pid < 0:
    print("fork failed")
    sys.exit(1)
elif pid == 0:
    sys.exit(0)
else:
    print("\n Process ID is: %d\n" % os.getpid())
    p.wait()
    sys.exit(0)

File: LABORATORY/laboratory1.cs
using System;
using System.Diagnostics;

class Program
{
    static void Main(string[] args)
    {
        int pid;

        Process p = new Process();
        p.StartInfo.FileName = "whoami";
        p.StartInfo.UseShellExecute = false;

        pid = p.Start() ? 1 : -1;

        if (pid < 0)
        {
            Console.WriteLine("fork failed");
        }
        else if (pid == 0)
        {
        }
        else
        {
            Console.WriteLine("\n Process ID is: {0}\n",
                Process.GetCurrentProcess().Id);

            p.WaitForExit();
        }
    }
}
